storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -358,6 +358,18 @@ func (s *peersSource) incrementalFlush(
 			NamespaceMetadata: nsMetadata,
 			Shard:             shard,
 			BlockStart:        start,
+			// If we've peer bootstrapped this shard/block combination AND the fileset
```

Comment:
Consider operator error where someone changes the kv config for bootstrappers to run peers before fs. That would violate the assumptions here. 

Could you include guards against allowing such configs to be used by a m3dbnode

Reply:
Added a guard for this in bootstrapper construction in `services/m3dbnode/config/bootstrapper.go`.

Reply:
Cheers